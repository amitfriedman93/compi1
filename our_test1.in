byte somebyte = 18b;

"also 'simple'"

"escape new lines\n"

"hex \x10"

"hex2 \x02"

"hex2 \x3A"

"hex3 \xff"