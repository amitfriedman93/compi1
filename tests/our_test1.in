byte somebyte = 18b;

"also 'simple'"

"escape new lines\n"





"hex2 \x3A"



void
int 
byte
b
bool
and
And
Bool
or
Or
OR
not
Not
NOT
nOT
true
True
TRUE
1
false
FALSE
0
False
return
RETURN
Return
ReTuRn
if
IF
If
iF
else
ELSE
Else
while
WHILE
While
break
BREAK
Break
continue
CONTINUE
Continue
;
,
(
)
{
}
=
==
!=
<
> 
<=
>=
+
-
*
/
// this is a comment
blah blah //commnt

x
max
OO7
12AB
42

// comment@
"hi\thow\tare\tyou"
"Hello \x57orld!\r\nThis\tis\t\x63oo\x6C, as always."
byte x = 15b;
print("Hello\nyou!");
int b = 15;
"te\0st"