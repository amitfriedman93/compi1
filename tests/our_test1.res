1 BYTE byte
1 ID somebyte
1 ASSIGN =
1 NUM 18
1 B b
1 SC ;
Error unclosed string
