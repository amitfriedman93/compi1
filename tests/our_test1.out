1 BYTE byte
1 ID somebyte
1 ASSIGN =
1 NUM 18
1 B b
1 SC ;
3 STRING also 'simple'
5 STRING escape new lines

11 STRING hex2 :
15 VOID void
16 INT int
17 BYTE byte
18 B b
19 BOOL bool
20 AND and
21 ID And
22 ID Bool
23 OR or
24 ID Or
25 ID OR
26 NOT not
27 ID Not
28 ID NOT
29 ID nOT
30 TRUE true
31 ID True
32 ID TRUE
33 NUM 1
34 FALSE false
35 ID FALSE
36 NUM 0
37 ID False
38 RETURN return
39 ID RETURN
40 ID Return
41 ID ReTuRn
42 IF if
43 ID IF
44 ID If
45 ID iF
46 ELSE else
47 ID ELSE
48 ID Else
49 WHILE while
50 ID WHILE
51 ID While
52 BREAK break
53 ID BREAK
54 ID Break
55 CONTINUE continue
56 ID CONTINUE
57 ID Continue
58 SC ;
59 COMMA ,
60 LPAREN (
61 RPAREN )
62 LBRACE {
63 RBRACE }
64 ASSIGN =
65 RELOP ==
66 RELOP !=
67 RELOP <
68 RELOP >
69 RELOP <=
70 RELOP >=
71 BINOP +
72 BINOP -
73 BINOP *
74 BINOP /
75 COMMENT //
76 ID blah
76 ID blah
76 COMMENT //
78 ID x
79 ID max
80 ID OO7
81 NUM 12
81 ID AB
82 NUM 42
84 COMMENT //
85 STRING hi	how	are	you
86 STRING Hello World!
This	is	cool, as always.
87 BYTE byte
87 ID x
87 ASSIGN =
87 NUM 15
87 B b
87 SC ;
88 ID print
88 LPAREN (
88 STRING Hello
you!
88 RPAREN )
88 SC ;
89 INT int
89 B b
89 ASSIGN =
89 NUM 15
89 SC ;
90 STRING te
